src/processor/operator/persistent/reader/csv/driver.cpp

```diff
@@ -29,20 +28,22 @@ void ParsingDriver::addValue(uint64_t rowNum, common::column_id_t columnIdx,
     BaseCSVReader* reader = getReader();
     if (columnIdx == reader->numColumns && length == 0) {
         // skip a single trailing delimiter in last columnIdx
-        return;
+        return true;
     }
     if (columnIdx >= reader->numColumns) {
-        throw CopyException(
-            stringFormat("Error in file {}, on line {}: expected {} values per row, but got more.",
-                reader->fileInfo->path, reader->getLineNumber(), reader->numColumns));
+        reader->handleCopyException(
+            stringFormat("expected {} values per row, but got more.", reader->numColumns));
+        return false;
     }
     try {
         function::CastString::copyStringToVector(chunk.getValueVector(columnIdx).get(), rowNum,
             value, &reader->option);
     } catch (ConversionException& e) {
-        throw CopyException(stringFormat("Error in file {} on line {}: {}", reader->fileInfo->path,
-            reader->getLineNumber(), e.what()));
+        reader->handleCopyException(e.what());
+        return false;
```

Comment:
Handling conversion exceptions using try/catch should have a pretty high performance penalty, keeping it this way in this PR for simplicity though (and it is still manageable in release mode). Just keep in mind that if there are a large number of conversion exceptions and we have backtraces enabled (e.g. in the tests) this will basically hang